src/main/java/edu/cornell/kfs/pdp/service/impl/CuCheckStubServiceImpl.java

```diff
@@ -0,0 +1,169 @@
+import org.kuali.kfs.kim.api.identity.PersonService;
+import org.kuali.kfs.kim.impl.identity.Person;
+import org.kuali.kfs.krad.bo.Note;
+import org.kuali.kfs.krad.document.Document;
+import org.kuali.kfs.krad.service.NoteService;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.purap.document.AccountsPayableDocumentBase;
+import org.kuali.kfs.module.purap.document.PaymentRequestDocument;
+import org.kuali.kfs.module.purap.document.VendorCreditMemoDocument;
+import org.kuali.kfs.pdp.businessobject.PaymentDetail;
+import org.kuali.kfs.pdp.businessobject.PaymentNoteText;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSConstants.PaymentSourceConstants;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.businessobject.RecurringDisbursementVoucherDetail;
+import edu.cornell.kfs.fp.document.CuDisbursementVoucherConstants;
+import edu.cornell.kfs.fp.document.RecurringDisbursementVoucherDocument;
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.pdp.CUPdpKeyConstants;
+import edu.cornell.kfs.pdp.service.CuCheckStubService;
+
+public class CuCheckStubServiceImpl implements CuCheckStubService {
+
+    private int checkStubMaxLengthForIso20022;
+    private ConfigurationService configurationService;
+    private NoteService noteService;
+    private PersonService personService;
+
+    @Override
+    public boolean doesCheckStubNeedTruncatingForIso20022(Document document) {
+        String paymentMethodCode;
+        if (ObjectUtils.isNull(document)) {
+            throw new IllegalArgumentException("document cannot be null");
+        } else if (document instanceof DisbursementVoucherDocument) {
+            paymentMethodCode = ((DisbursementVoucherDocument) document).getPaymentMethodCode();
+            if (document instanceof RecurringDisbursementVoucherDocument
+                    && doAnyRecurringCheckStubsNeedTruncatingForIso20022(
+                            (RecurringDisbursementVoucherDocument) document)) {
+                return true;
+            }
+        } else if (document instanceof PaymentRequestDocument) {
+            paymentMethodCode = ((PaymentRequestDocument) document).getPaymentMethodCode();
+        } else if (document instanceof VendorCreditMemoDocument) {
+            paymentMethodCode = ((VendorCreditMemoDocument) document).getPaymentMethodCode();
+        } else {
+            throw new IllegalArgumentException("document of type " + document.getClass().getSimpleName()
+                    + " does not have support for both check stubs and payment methods");
+        }
+        return StringUtils.equalsIgnoreCase(paymentMethodCode, PaymentSourceConstants.PAYMENT_METHOD_CHECK)
+                && StringUtils.length(getFullCheckStub(document)) > checkStubMaxLengthForIso20022;
+    }
+
+    private boolean doAnyRecurringCheckStubsNeedTruncatingForIso20022(RecurringDisbursementVoucherDocument document) {
```

Comment:
This function may need some restructuring to help with clarity.  I had to read it a couple of times.  Just to be sure I think the code should only be processed if the recurring DV has recurring payment details, and the payment method of the RCDV is check.
Then the code should loop through each check stub text to make sure it isn't too long

Reply:
Yes, that's basically what this method does. I'll see if I can rewrite it to make the intent clearer.